Manual navigation tracking for Android (#1465)
splunk/public-o11y-docs: gdi/get-data-in/rum/android/manual-rum-android-instrumentation.rst

      ...
   }

.. _android-rum-manual-navigation-tracking:

Manually track navigation events
=====================================

By default, the Android RUM agent follows the view lifecycle of ``Fragment`` and ``Activity`` instances, and certain UI frameworks, e.g. Jetpack Compose, do not use that view lifecycle. In that case, starting with the version 1.6.0, :code:`experimentalSetScreenName()` can be used to explitly signal that navigation has occurred. Method name is prefixed with :code:`experimental` to denote that this API might change in the future, possibly even between minor releases.

In general, upon a non-``Fragment``, non-``Activity`` navigation event, the application developer calls:

.. code-block:: java

   SplunkRum.getInstance().experimentalSetScreenName(screenName);

> which both sends a navigation span to RUM, and remembers the screen name which will then be assigned as an attribute on the later spans.

Reviewer: ```suggestion
This sends a navigation span to RUM and remembers the screen name for subsequent spans.
```

> Once the explicit screen name is set, it overrides the default view livecycle tracking. If your application consists of both ``Activity`` and non-``Activity`` views, then upon exiting the view with non-``Activity`` navigation, the explicit screen name must be cleared:

Reviewer: ```suggestion
Once the explicit screen name is set, it overrides the default view livecycle tracking. If your application consists of both ``Activity`` and non-``Activity`` views, you must clear the explicit screen name when exiting the non-``Activity`` view:
```

.. code-block:: java

   // doubled to clear both the last view and the previous last view
   SplunkRum.getInstance().experimentalSetScreenName(null)
   SplunkRum.getInstance().experimentalSetScreenName(null)

And if you'd like to track these in RUM, ``Restarted`` and ``Resumed`` events must then also be signalled explicitly, for example:

.. code-block:: kotlin

   override fun onResume() {
      super.onResume()
      // in this case you need to store lastScreen yourself
      if (lastScreen != null) {

>          SplunkRum.getInstance().experimentalSetScreenName(lastScreen, "Resumed")

Reviewer: I'm hesitant to get into this level of detail. Do we have these constants documented anywhere else? They are very likely to change as upstream otel semantic conventions evolve, and I think we're doing our customers a disservice by recommending that they use bespoke values.

I know it leaves part of the picture incomplete, but I think I'd prefer to drop this section about Restarted and Resumed.

Reviewer 2: I deleted that section
